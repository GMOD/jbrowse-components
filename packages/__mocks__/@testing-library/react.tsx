import React, { Suspense } from 'react'
import { createTheme, ThemeProvider } from '@mui/material/styles'

const react = jest.requireActual('@testing-library/react')
const render = (args: React.ReactNode) => {
  return react.render(
    <Suspense fallback={null}>
      <ThemeProvider theme={createTheme()}>{args}</ThemeProvider>
    </Suspense>,
  )
}

module.exports = { ...react, render }
