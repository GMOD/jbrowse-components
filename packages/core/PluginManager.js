import React from 'react'
import * as mst from 'mobx-state-tree'

import RendererType from './pluggableElementTypes/renderers/RendererType'
import AdapterType from './pluggableElementTypes/AdapterType'
import TrackType from './pluggableElementTypes/TrackType'
import ViewType from './pluggableElementTypes/ViewType'
import DrawerWidgetType from './pluggableElementTypes/DrawerWidgetType'
import MenuBarType from './pluggableElementTypes/MenuBarType'
import ConnectionType from './pluggableElementTypes/ConnectionType'

import { ConfigurationSchema } from './configuration'

// little helper class that keeps groups of callbacks that are
// then run in a specified order by group
class PhasedScheduler {
  phaseCallbacks = {}

  constructor(...phaseOrder) {
    this.phaseOrder = phaseOrder
  }

  add(phase, callback) {
    if (!this.phaseOrder.includes(phase)) {
      throw new Error(`unknown phase ${phase}`)
    }
    if (!this.phaseCallbacks[phase]) this.phaseCallbacks[phase] = []
    this.phaseCallbacks[phase].push(callback)
  }

  run() {
    this.phaseOrder.forEach(phaseName => {
      if (this.phaseCallbacks[phaseName]) {
        this.phaseCallbacks[phaseName].forEach(callback => callback())
      }
    })
  }
}

export default class PluginManager {
  plugins = []

  elementTypes = {}

  elementCreationSchedule = new PhasedScheduler(
    'renderer',
    'adapter',
    'track',
    'connection',
    'view',
    'drawer widget',
    'menu bar',
  )

  typeBaseClasses = {
    renderer: RendererType,
    adapter: AdapterType,
    track: TrackType,
    connection: ConnectionType,
    view: ViewType,
    'drawer widget': DrawerWidgetType,
    'menu bar': MenuBarType,
  }

  static lib = { 'mobx-state-tree': mst, React }

  constructor(initialPlugins = []) {
    this.lib = PluginManager.lib

    this.getRendererType = this.getElementType.bind(this, 'renderer')
    this.getAdapterType = this.getElementType.bind(this, 'adapter')
    this.getTrackType = this.getElementType.bind(this, 'track')
    this.getViewType = this.getElementType.bind(this, 'view')
    this.getDrawerWidgetType = this.getElementType.bind(this, 'drawer widget')
    this.getMenuBarType = this.getElementType.bind(this, 'menu bar')
    this.getConnectionType = this.getElementType.bind(this, 'connection')
    this.addRendererType = this.addElementType.bind(this, 'renderer')
    this.addAdapterType = this.addElementType.bind(this, 'adapter')
    this.addTrackType = this.addElementType.bind(this, 'track')
    this.addViewType = this.addElementType.bind(this, 'view')
    this.addDrawerWidgetType = this.addElementType.bind(this, 'drawer widget')
    this.addMenuBarType = this.addElementType.bind(this, 'menu bar')
    this.addConnectionType = this.addElementType.bind(this, 'connection')

    // add all the initial plugins
    initialPlugins.forEach(plugin => {
      this.addPlugin(plugin)
    })
  }

  addPlugin(plugin) {
    if (this.configured) {
      throw new Error('JBrowse already configured, cannot add plugins')
    }
    if (this.plugins.includes(plugin)) {
      throw new Error('plugin already installed')
    }
    // if (!plugin.install) console.error(plugin)
    plugin.install(this)
    this.plugins.push(plugin)
    return this
  }

  /** get a MST type for the union of all specified pluggable MST types */
  pluggableMstType(
    typeGroup,
    fieldName,
    fallback = mst.types.maybe(mst.types.null),
  ) {
    const pluggableTypes = this.getElementTypeMembers(typeGroup, fieldName)
    // try to smooth over the case when no types are registered, mostly encountered in tests
    if (pluggableTypes.length === 0) return fallback
    return mst.types.union(...pluggableTypes)
  }

  /** get a MST type for the union of all specified pluggable config schemas */
  pluggableConfigSchemaType(
    typeGroup,
    fieldName = 'configSchema',
    fallback = ConfigurationSchema('Null', {}),
  ) {
    return this.pluggableMstType(typeGroup, fieldName, fallback)
  }

  addElementType(groupName, creationCallback) {
    if (typeof creationCallback !== 'function') {
      throw new Error('must provide a callback function')
    }
    const typeBaseClass = this.typeBaseClasses[groupName]
    if (!typeBaseClass) {
      throw new Error(`unknown pluggable element type ${groupName}, cannot add`)
    }
    if (!this.elementTypes[groupName]) this.elementTypes[groupName] = {}

    this.elementCreationSchedule.add(groupName, () => {
      const element = creationCallback(this)
      if (
        groupName === 'adapter' &&
        !element.AdapterClass.capabilities.length
      ) {
        throw new Error(
          `Adapter ${
            element.AdapterClass.name
          } must provide a static property "capabilities" that has at least one entry. See BaseAdapter for an example.`,
        )
      }
      if (this.elementTypes[groupName][element.name]) {
        throw new Error(
          `${groupName} ${
            element.name
          } already registered, cannot register it again`,
        )
      }

      this.elementTypes[groupName][element.name] = element
    })

    return this
  }

  getElementType(groupName, typeName) {
    if (!typeName) {
      throw new Error(`invalid type name "${typeName}"`)
    }
    return (this.elementTypes[groupName] || {})[typeName]
  }

  getElementTypesInGroup(groupName) {
    return Object.values(this.elementTypes[groupName] || {})
  }

  getElementTypeMembers(groupName, memberName) {
    return this.getElementTypesInGroup(groupName)
      .map(t => t[memberName])
      .filter(m => !!m)
  }

  configure() {
    if (this.configured) throw new Error('already configured')

    // run the creation callbacks for each element type in order.
    // see elementCreationSchedule above for the creation order
    this.elementCreationSchedule.run()
    delete this.elementCreationSchedule

    this.plugins.forEach(plugin => plugin.configure())

    this.configured = true

    // console.log(JSON.stringify(getSnapshot(model)))

    return this
  }
}
