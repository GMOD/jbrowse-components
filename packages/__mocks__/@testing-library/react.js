import { StylesProvider } from '@material-ui/core/styles'
import React from 'react'

const react = jest.requireActual('@testing-library/react')
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`
const render = args => {
  return react.render(
    <StylesProvider generateClassName={generateClassName}>
      {args}
    </StylesProvider>,
  )
}

module.exports = { ...react, render }
