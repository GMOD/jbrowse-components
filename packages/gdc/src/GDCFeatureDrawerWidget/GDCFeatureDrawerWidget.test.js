import React from 'react'
import { render } from '@testing-library/react'
import { observable } from 'mobx'

import PluginManager from '@gmod/jbrowse-core/PluginManager'
import GDCFeatureDetailsF from './GDCFeatureDrawerWidget'

const pm = new PluginManager([])
const GDCFeatureDetails = pm.load(GDCFeatureDetailsF)

describe('GDCTrack drawer widget', () => {
  it('renders mutation with just the required model elements', () => {
    const f = observable({
      featureData: {
        uniqueId: '0208efeb-f1e8-57e4-8447-299c5f050380',
        refName: 'chr3',
        type: 'Simple Somatic Mutation',
        start: 377917,
        end: 377918,
        chromosome: 'chr3',
        consequence: {
          hits: {
            edges: [
              {
                node: {
                  id:
                    'U1NNQ29uc2VxdWVuY2U6MDIwOGVmZWItZjFlOC01N2U0LTg0NDctMjk5YzVmMDUwMzgwOjZkYmQ5M2M2LWYwZWYtNTdhZS1iZmQxLWYxM2RlNmM5ZWI0Ng==',
                  transcript: {
                    aa_change: 'D618N',
                    annotation: {
                      hgvsc: 'c.1852G>A',
                      polyphen_impact: 'benign',
                      polyphen_score: 0.008,
                      sift_impact: 'tolerated',
                      sift_score: 0.08,
                      vep_impact: 'MODERATE',
                    },
                    consequence_type: 'missense_variant',
                    gene: {
                      gene_id: 'ENSG00000134121',
                      gene_strand: 1,
                      symbol: 'CHL1',
                    },
                    is_canonical: false,
                    transcript_id: 'ENST00000620033',
                  },
                },
              },
            ],
          },
        },
        cosmicId: ['COSM1044638'],
        endPosition: 377918,
        genomicDnaChange: 'chr3:g.377918G>A',
        mutationSubtype: 'Single base substitution',
        mutationType: 'Simple Somatic Mutation',
        ncbiBuild: 'GRCh38',
        referenceAllele: 'G',
        score: 3,
        // Prevent from trying project fetch
        // ssmId: '0208efeb-f1e8-57e4-8447-299c5f050380',
        startPosition: 377918,
      },
    })
    /* eslint-enable @typescript-eslint/camelcase */

    const { container } = render(<GDCFeatureDetails model={f} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders gene with just the required model elements', () => {
    const f = observable({
      featureData: {
        uniqueId: 'ENSG00000134121',
        refName: '3',
        type: 'protein_coding',
        start: 196595,
        end: 409417,
        biotype: 'protein_coding',
        canonicalTranscriptId: 'ENST00000256509',
        description:
          'The protein encoded by this gene is a member of the L1 gene family of neural cell adhesion molecules. It is a neural recognition molecule that may be involved in signal transduction pathways. The deletion of one copy of this gene may be responsible for mental defects in patients with 3p- syndrome. This protein may also play a role in the growth of certain cancers. Alternate splicing results in both coding and non-coding variants. [provided by RefSeq, Nov 2011]',
        externalDbIds: {
          entrezGene: ['10752'],
          hgnc: ['HGNC:1939'],
          omimGene: ['607416'],
          uniprotkbSwissprot: ['O00533'],
        },
        geneChromosome: '3',
        geneEnd: 409417,
        // Prevent from trying project fetch
        // geneId: 'ENSG00000134121',
        geneStart: 196596,
        geneStrand: 1,
        id:
          'R2VuZTpFTlNHMDAwMDAxMzQxMjEjZTQ2MDk1NGM4MmZiMjczYjIwNTk1MDliODFkZTZmYjEj',
        isCancerGeneCensus: null,
        name: 'cell adhesion molecule L1-like',
        symbol: 'CHL1',
        synonyms: ['CALL', 'FLJ44930', 'L1CAM2', 'MGC132578'],
      },
    })

    const { container } = render(<GDCFeatureDetails model={f} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
