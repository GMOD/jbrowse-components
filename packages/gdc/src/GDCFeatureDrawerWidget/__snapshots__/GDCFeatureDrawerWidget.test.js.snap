// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GDCTrack drawer widget renders gene with just the required model elements 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="variant-side-drawer"
>
  <div
    class="MuiPaper-root makeStyles-paperRoot MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="alignment-side-drawer"
  >
    <div
      class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-expanded MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-top: 4px;"
    >
      <div
        aria-disabled="false"
        aria-expanded="true"
        class="MuiButtonBase-root MuiExpansionPanelSummary-root MuiExpansionPanelSummary-expanded"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiExpansionPanelSummary-content MuiExpansionPanelSummary-expanded"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
             
            Primary data
          </span>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiExpansionPanelSummary-expanded MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root makeStyles-expandIcon"
            >
              expand_more
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-entered"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiExpansionPanelDetails-root makeStyles-expansionPanelDetails"
              >
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Position
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      3:196596..409417 
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Description
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      The protein encoded by this gene is a member of the L1 gene family of neural cell adhesion molecules. It is a neural recognition molecule that may be involved in signal transduction pathways. The deletion of one copy of this gene may be responsible for mental defects in patients with 3p- syndrome. This protein may also play a role in the growth of certain cancers. Alternate splicing results in both coding and non-coding variants. [provided by RefSeq, Nov 2011]
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Name
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      cell adhesion molecule L1-like
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Length
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      212822
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Type
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      protein_coding
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-expanded MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-top: 4px;"
    >
      <div
        aria-disabled="false"
        aria-expanded="true"
        class="MuiButtonBase-root MuiExpansionPanelSummary-root MuiExpansionPanelSummary-expanded"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiExpansionPanelSummary-content MuiExpansionPanelSummary-expanded"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
             
            Attributes
          </span>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiExpansionPanelSummary-expanded MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root makeStyles-expandIcon"
            >
              expand_more
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-entered"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiExpansionPanelDetails-root makeStyles-expansionPanelDetails"
              >
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    biotype
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      protein_coding
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    description
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      The protein encoded by this gene is a member of the L1 gene family of neural cell adhesion molecules. It is a neural recognition molecule that may be involved in signal transduction pathways. The deletion of one copy of this gene may be responsible for mental defects in patients with 3p- syndrome. This protein may also play a role in the growth of certain cancers. Alternate splicing results in both coding and non-coding variants. [provided by RefSeq, Nov 2011]
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    geneChromosome
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      3
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    geneEnd
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      409417
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    geneStart
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      196596
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    geneStrand
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      1
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    id
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      R2VuZTpFTlNHMDAwMDAxMzQxMjEjZTQ2MDk1NGM4MmZiMjczYjIwNTk1MDliODFkZTZmYjEj
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    isCancerGeneCensus
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      null
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    symbol
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      CHL1
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    synonyms
                  </div>
                  <div
                    class="makeStyles-fieldSubvalue"
                  >
                    <div>
                      CALL
                    </div>
                  </div>
                  <div
                    class="makeStyles-fieldSubvalue"
                  >
                    <div>
                      FLJ44930
                    </div>
                  </div>
                  <div
                    class="makeStyles-fieldSubvalue"
                  >
                    <div>
                      L1CAM2
                    </div>
                  </div>
                  <div
                    class="makeStyles-fieldSubvalue"
                  >
                    <div>
                      MGC132578
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="MuiDivider-root"
  />
  <hr
    class="MuiDivider-root"
  />
  <hr
    class="MuiDivider-root"
  />
</div>
`;

exports[`GDCTrack drawer widget renders mutation with just the required model elements 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="variant-side-drawer"
>
  <div
    class="MuiPaper-root makeStyles-paperRoot MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="alignment-side-drawer"
  >
    <div
      class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-expanded MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-top: 4px;"
    >
      <div
        aria-disabled="false"
        aria-expanded="true"
        class="MuiButtonBase-root MuiExpansionPanelSummary-root MuiExpansionPanelSummary-expanded"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiExpansionPanelSummary-content MuiExpansionPanelSummary-expanded"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
             
            Primary data
          </span>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiExpansionPanelSummary-expanded MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root makeStyles-expandIcon"
            >
              expand_more
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-entered"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiExpansionPanelDetails-root makeStyles-expansionPanelDetails"
              >
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Position
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      chr3:377918..377918 
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Length
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      1
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Type
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      Simple Somatic Mutation
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-expanded MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-top: 4px;"
    >
      <div
        aria-disabled="false"
        aria-expanded="true"
        class="MuiButtonBase-root MuiExpansionPanelSummary-root MuiExpansionPanelSummary-expanded"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiExpansionPanelSummary-content MuiExpansionPanelSummary-expanded"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
             
            Attributes
          </span>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiExpansionPanelSummary-expanded MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root makeStyles-expandIcon"
            >
              expand_more
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-entered"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiExpansionPanelDetails-root makeStyles-expansionPanelDetails"
              >
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    chromosome
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      chr3
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    endPosition
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      377918
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    genomicDnaChange
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      chr3:g.377918G&gt;A
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    mutationSubtype
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      Single base substitution
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    mutationType
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      Simple Somatic Mutation
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    ncbiBuild
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      GRCh38
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    referenceAllele
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      G
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    score
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      3
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    startPosition
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      377918
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="MuiDivider-root"
  />
  <hr
    class="MuiDivider-root"
  />
  <hr
    class="MuiDivider-root"
  />
</div>
`;
